src/main/java/com/kilopo/kosshop/service/Impl/ProductServiceImpl.java

```diff
@@ -22,6 +22,6 @@ public boolean add(Product product) {
     public List<Product> searchByFields(Map<String, String> mapProductParam) {
         Map<String, String> filteredmap = mapProductParam.entrySet().stream().filter(mapEntry -> !mapEntry.getValue()
                 .equals("")).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-        return productDAO.findByFields(filteredmap);
+        return filteredmap.isEmpty() ? null : productDAO.findByFields(filteredmap);
```

Comment:
good practice to return empty list. not null
with this approach you can avoid NullPointerException